mxnet-bot/EmailBot/EmailBot.py

```diff
@@ -0,0 +1,382 @@
+        # generate the second html tabel
+        htmltable2 = [["<a href='" +"https://github.com/{}/issues/{}".format(self.repo,str(item['number']) ) + "'>" + str(item['number']) + "</a>   ", 
+                       ",".join(item['predictions'])] for item in self.predict()]
+        body_html = """<html>
+        <head>
+        </head>
+        <body>
+          <h4>Week: {} to {}</h4>
+          <p>{} newly issues were opened in the above period, among which {} were closed and {} are still open.</p>
+          <div>{}</div>
+          <p>Here are the recommanded labels for unlabeled issues:</p>
+          <div>{}</div>
+          <p><img src="cid:image1" width="400" height="400"></p>
+        </body>
+        </html>
+                    """.format(str(self.start.date()), str((self.end - datetime.timedelta(days=2)).date()),
+                               str(len(self.open_issues) + len(self.closed_issues)),
+                               str(len(self.closed_issues)), str(len(self.open_issues)),
+                               "\n".join(self.__html_table(htmltable)),
+                               "\n".join(self.__html_table(htmltable2)))
+        return body_html
+
+    def sendemail(self):
+        """
+        This method is to send emails.
+        The email content contains 2 html tables and an image.
+        """
+        sender = self.sender
+        recipients = self.recipients
+        aws_region = self.aws_region
+        # The email body for recipients with non-HTML email clients.
+        body_text = "weekly report"
+        # The HTML body of the email.
+        body_html = self.__bodyhtml()
+        # The subject line for the email.
+        subject = "GitHub Issues Daily Report {} to {}".format(str(self.start.date()),
+                                                               str((self.end - datetime.timedelta(days=2)).date()))
+        # The character encoding for the email.
+        charset = "utf-8"
+        # Create a new SES resource and specify a region.
+        client = boto3.client('ses', region_name=aws_region)
+
+        # Create a multipart/mixed parent container.
+        msg = MIMEMultipart('mixed')
+        # Add subject, from and to lines
+        msg['Subject'] = subject
+        msg['From'] = sender
+        msg['To'] = ",".join(recipients)
+
+        # Create a multiparter child container
+        msg_body = MIMEMultipart('alternative')
+
+        # Encode the text and HTML content and set the character encoding. This step is
+        # necessary if you're sending a message with characters outside the ASCII range.
+        textpart = MIMEText(body_text.encode(charset), 'plain', charset)
+        htmlpart = MIMEText(body_html.encode(charset), 'html', charset)
+
+        # Add the text and HTML parts to the child container
+        msg_body.attach(textpart)
+        msg_body.attach(htmlpart)
+        msg.attach(msg_body)
+
+        # Attach Image
+        fg = open(self.img_file, 'rb')
```

Comment:
can we use with open(....) as f:   It's the safe way to deal with exceptions and errors and not leak open files.